Public/Src/Engine/Scheduler/PipExecutor.cs

```diff
@@ -3659,6 +3660,18 @@ internal static void UpdateFileData(Dictionary<AbsolutePath, FileOutputData> map
                                 continue;
                             }
 
+                            if (!OperatingSystemHelper.IsUnixOS)
+                            {
+                                var pathstring = access.ToString(pathTable);
+                                if (Directory.Exists(pathstring))
+                                {
+                                    // We run into here becuase maybeResult.Result != PathExistence.ExistsAsDirectory
+                                    // But the path is actually a Directory, so it is a dir symlink or junction.
+                                    // We don't support storing directory symlinks/junctions to the cache in Windows right now.
+                                    continue;

```

Comment:
>continue; [](start = 36, length = 9)

There are two choices here:
- We log error, stating that we don't support directory symlink production.
- We log warning, but make the pip uncacheable.